infra/es_cluster.tf

```diff
@@ -208,10 +209,10 @@ resource "google_compute_instance" "es-init" {
   }
 
   service_account {
-    email = google_service_account.es.email
+    email = google_service_account.es-discovery.email
     scopes = [
       # Required for cloud logging
-      "cloud-platform",
+      "logging-write",
```

Comment:
is that less or more permissions or just different and the old one didn’t work?

Reply:
It's less, but it's also unchanged. GCP has this great notion of having two different permission systems for instances: at one level, you specify "scopes", which is what we're doing here. This is the "old", deprecated-but-you-still-have-to-use-it way. The official recommendation (from GCP) here is to set it to "cloud-platform", which means "access to everything".

The "current" way of specifying permissions is via the service account associated with the instance.

In order for an action to succeed it needs to be permitted by both the scopes and the service account (which makes debugging permission issues a bit painful at times). So this is restricting the scope, but we were already (and still are) fully restricted to the absolute minimum set of rights by the service account, which is why "less, but actually the same".